actors/market/tests/publish_storage_deals_failures.rs

```diff
@@ -203,4 +208,268 @@ mod publish_storage_deals_failures {
         };
         assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
     }
+
+    #[test]
+    fn fail_when_client_has_some_funds_but_not_enough_for_a_deal() {
+        let mut rt = setup();
+
+        //
+        add_participant_funds(&mut rt, CLIENT_ADDR, TokenAmount::from(100u8));
+        let start_epoch = 42;
+        let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+        let deal1 = generate_deal_proposal(CLIENT_ADDR, PROVIDER_ADDR, start_epoch, end_epoch);
+        add_provider_funds(&mut rt, deal1.clone().provider_collateral, &MinerAddresses::default());
+        let buf = RawBytes::serialize(deal1.clone()).expect("failed to marshal deal proposal");
+        let sig = Signature::new_bls("does not matter".as_bytes().to_vec());
+        let params = PublishStorageDealsParams {
+            deals: vec![ClientDealProposal {
+                proposal: deal1.clone(),
+                client_signature: sig.clone(),
+            }],
+        };
+
+        rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
```

Comment:
nit: this LGTM but for the next two tests consider using @LesnyRumcajs's nice utility just introduced [here](https://github.com/filecoin-project/builtin-actors/pull/319/files#diff-6ad2f8c7f5d0122a8e0196c96b95fb4ebbc8c3f977f134c2c872f7723e21b328R487)